auraed/tests/vms_start_must_start_vm_with_auraed.rs

```diff
@@ -1,61 +1,61 @@
-//     //                     host_path: "/dev/vda1".to_string(),
-//     //                     is_writeable: true,
-//     //                 }),
-//     //                 drive_mounts: vec![DriveMount {
-//     //                     host_path: "/var/lib/aurae/vm/image/disk.raw"
-//     //                         .to_string(),
-//     //                     vm_path: "/".to_string(),
-//     //                     fs_type: "ext4".to_string(),
-//     //                     is_writeable: true,
-//     //                 }],
-//     //             }),
-//     //         })
-//     //         .await
-//     // );
-//
-//     // assert!(res.is_ok());
-//
-//     // let res = retry!(
-//     //     client.start(VmServiceStartRequest { vm_id: vm_id.clone() }).await
-//     // );
-//
-//     // assert!(res.is_ok());
-//
-//     // NOTE: for now this passes when cloud-hypervisor is running a VM with auraed
-//     //       as PID 1 with a tuntap device at the provided scope_id (4 on jeroen's machine)
-//     // let remote_client =
-//     //     common::remote_auraed_client("[fe80::2]:8080".to_string(), 4).await;
-//     // let res = remote_client.discover(DiscoverRequest {}).await;
-//     //
-//     // assert!(res.is_ok());
-//     //
-//     // let res = res.expect("this shouldn't happen").into_inner();
-//     // assert!(res.healthy);
-// }
+use std::thread;
+
+use client::discovery::discovery_service::DiscoveryServiceClient;
+use client::vms::vms_service::VmServiceClient;
+use proto::discovery::DiscoverRequest;
+use proto::vms::{
+    RootDrive, VirtualMachine, VmServiceCreateRequest, VmServiceStartRequest,
+};
+
+mod common;
+
+#[test_helpers_macros::shared_runtime_test]
+async fn vms_must_start_vm_with_auraed() {
+    let vm_id = format!("ae-test-vm-{}", uuid::Uuid::new_v4());
+    let client = common::auraed_client().await;
+    let res = retry!(
+        client
+            .create(VmServiceCreateRequest {
+                machine: Some(VirtualMachine {
+                    id: vm_id.clone(),
+                    mem_size_mb: 1024,
+                    vcpu_count: 2,
+                    kernel_img_path: "/var/lib/aurae/vm/kernel/vmlinux.bin"
+                        .to_string(),
+                    kernel_args: vec![
+                        "console=hvc0".to_string(),
+                        "root=/dev/vda1".to_string(),
+                        "rw".to_string(),
+                    ],
+                    root_drive: Some(RootDrive {
+                        host_path: "/var/lib/aurae/vm/image/disk.raw".into(),
+                        is_writeable: true,
+                    }),
+                    drive_mounts: vec![],
+                }),
+            })
+            .await
+    );
+
+    assert!(res.is_ok(), "{:?}", res);
+
+    let res = retry!(
+        client.start(VmServiceStartRequest { vm_id: vm_id.clone() }).await
+    );
+
+    assert!(res.is_ok(), "{:?}", res);
+    thread::sleep(std::time::Duration::from_secs(10));
```

Comment:
it would be nice to not need this sleep.. is there a signal we can get (in future) from the health or discovery service  to poll?

Reply:
Absolutely - I'll add an endpoint to list VMs and use that combined with the VMs status field to know when to attempt to build the client and connect to the VMs Auraed.

Reply:
Took a stab at something in [f08afa7](https://github.com/aurae-runtime/aurae/pull/506/commits/f08afa77ab8738229d7daa80291be07d581dca2c). 

I attempted to address both polling the VM's status and fetching the nested `auraed` socket address to facilitate connecting to it.